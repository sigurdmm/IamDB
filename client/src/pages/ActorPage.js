import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchActorById } from '../modules/media/actions';
import ActorInformation from '../components/ActorInformation';
import CoverDisplay from '../components/CoverDisplay/index';
import './FilmPage.less';

export class ActorPage extends React.Component {
  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string.isRequired,
      }),
    }),
    fetchActorById: PropTypes.func.isRequired,
    detailedActor: PropTypes.shape({
      id: PropTypes.any.isRequired,
      name: PropTypes.string.isRequired,
      popularity: PropTypes.number,
      media: PropTypes.array,
      thumbnails: PropTypes.shape({
        small: PropTypes.string,
      }),
    }),
    loading: PropTypes.bool,
    error: PropTypes.string,
  };

  componentDidMount() {
    this.props.fetchActorById(this.props.match.params.id);
  }

  render() {
    const { detailedActor, loading, error } = this.props;

    if (loading) {
      return <div>
        Laster inn
      </div>;
    }

    if (error) {
      return <div className='error__container'>
        Kunne ikke finne informasjon
      </div>;
    }

    return <div className='media__container'>
      <ActorInformation details={detailedActor}/>
      <div>
        <h2>Known for</h2>
        <CoverDisplay hasSearched={true} media={detailedActor.media} url='/media/'/>
      </div>
    </div>;
  }
}

const mapStateToProps = state => ({
  detailedActor: state.media.detailedActor,
  loading: state.media.loading,
  error: state.media.error,
});

const actionsToProps = {
  fetchActorById,
};

export default connect(mapStateToProps, actionsToProps)(ActorPage);
