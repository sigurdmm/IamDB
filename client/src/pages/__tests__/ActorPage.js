import { MemoryRouter } from 'react-router-dom';
import { mount, render } from 'enzyme';

// Note we're not using the default exported version
import { ActorPage } from '../ActorPage';

const mockedDetails = {
  id: -1,
  name: '',
  thumbnails: {},
  media: [{
    name: 'movie',
    director: 'director',
  }],
  popularity: 0,
};

describe('<ActorPage/>', () => {
  it('should render properly', () => {
    const mockedDetailsNoDirector = Object.assign(mockedDetails, {});
    // Remove directors
    mockedDetailsNoDirector.media.map(m => Object.assign(m, { director: null }));

    const spy = jest.fn();
    const wrapper = render(<MemoryRouter>
      <ActorPage
        fetchActorById={spy}
        detailedActor={mockedDetailsNoDirector}
        loading={false}
        error={null}
      />
    </MemoryRouter>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should render properly when loading', () => {
    const spy = jest.fn();
    const wrapper = render(<MemoryRouter><ActorPage
      fetchActorById={spy}
      detailedActor={mockedDetails}
      loading={true}
      error={null}
    /></MemoryRouter>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render properly when error', () => {
    const spy = jest.fn();
    const wrapper = render(<MemoryRouter><ActorPage
      fetchActorById={spy}
      detailedMedia={mockedDetails}
      loading={false}
      error={true}
    /></MemoryRouter>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should fetch detailedActor', () => {
    const id = 1337;
    const spy = jest.fn();
    mount(<MemoryRouter><ActorPage
      fetchActorById={spy}
      match={{
        params: { id },
      }}
      detailedActor={mockedDetails}
      loading={false}
      error={null}
    /></MemoryRouter>);
    expect(spy).toHaveBeenCalled();
    expect(spy).toHaveBeenCalledWith(id);
  });
});
