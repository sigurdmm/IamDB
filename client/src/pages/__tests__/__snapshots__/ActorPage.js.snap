// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActorPage/> should render properly 1`] = `
<div
  class="media__container"
>
  <div
    class="information__container"
  >
    <div
      class="mediaimage"
    >
      <img
        class="image__item"
        src="no-image-found.jpg"
      />
    </div>
    <div>
      <div
        class="title__container"
      >
        <h2 />
      </div>
      <h3>
        ⭐️: 0
      </h3>
    </div>
  </div>
  <section>
    <h2>
      Known for
    </h2>
    <div
      class="coverdisplay"
    />
  </section>
  <section>
    <h2>
      Directors  has worked with
    </h2>
    <div
      class="wordcloud"
    >
      <div>
        <svg
          height="250"
          width="400"
        >
          <g
            transform="translate(200,125)"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;

exports[`<ActorPage/> should render properly when error 1`] = `
<div
  class="error__container"
>
  Kunne ikke finne informasjon
</div>
`;

exports[`<ActorPage/> should render properly when loading 1`] = `
<div>
  Laster inn
</div>
`;
